contracts/access/manager/AuthorityUtils.sol

```diff
@@ -17,16 +18,21 @@ library AuthorityUtils {
         address target,
         bytes4 selector
     ) internal view returns (bool immediate, uint32 delay) {
-        (bool success, bytes memory data) = authority.staticcall(
-            abi.encodeCall(IAuthority.canCall, (caller, target, selector))
-        );
-        if (success) {
-            if (data.length >= 0x40) {
-                (immediate, delay) = abi.decode(data, (bool, uint32));
-            } else if (data.length >= 0x20) {
-                immediate = abi.decode(data, (bool));
+        Memory.FreePtr ptr = Memory.save();
+
+        bytes memory params = abi.encodeCall(IAuthority.canCall, (caller, target, selector));
+        assembly ("memory-safe") {
+            let success := staticcall(not(0), authority, add(params, 0x20), mload(params), 0, 0x40)
+            if success {
+                if gt(returndatasize(), 0x1F) {
+                    immediate := gt(mload(0x00), 0)
+                }
+                if gt(returndatasize(), 0x3F) {
+                    delay := and(mload(0x20), 0xFFFFFFFF)
```

Comment:
Should this revert if delay is too large ?
```suggestion
                    delay := mload(0x20)
                    if gt(delay, 0xFFFFFFFF) { revert(0,0) }
```

Reply:
`abi.decode` reverts, so I guess we should revert too to keep backwards compatibility